### PR TITLE
Add CUDA+clang CI test

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: test/ArborX_EnableDeviceTypes.hpp.in [hunk 1]
```diff
@@ -1,13 +1,10 @@
 #ifndef ARBORX_ENABLE_DEVICE_TYPES_HPP
 #define ARBORX_ENABLE_DEVICE_TYPES_HPP
 
-#include <Kokkos_Macros.hpp>
-
-#include <boost/utility/identity_type.hpp>
+#include <Kokkos_Core.hpp>
```
Comment (dalg24) [inline test/ArborX_EnableDeviceTypes.hpp.in]: Do we need the Kokkos header at all here?

Comment (masterleinad) [reply test/ArborX_EnableDeviceTypes.hpp.in]: We need the definitions for the device types.

Comment (dalg24) [reply test/ArborX_EnableDeviceTypes.hpp.in]: I was going to say no you don't but now I realize you traded the macro for an alias.

Comment (masterleinad) [reply test/ArborX_EnableDeviceTypes.hpp.in]: I could try to see if reverting that still works if you want. In the end, we have to include that header anyway, though, and I prefer using as little macros as necessary.

**Comment by dalg24** [comment]
Rebase on master and amended Kokkos_ENABLE_MPI -> KOKKOS_ENABLE_MPI

**Comment by dalg24** [comment]
Nobody is allowed to force-push over me :@
Amended mixup into initial commit.

**Comment by masterleinad** [comment]
Thanks!

**Pull request merged**